docs/writing-rules/data-flow/taint-mode.md

```diff
@@ -182,118 +500,124 @@ pattern-propagators:
 
-A sink is specified by a pattern but, unlike for sources and sanitizers, Semgrep finds the best match for the sink.
+## Minimizing false positives
 
-### Function argument as a sink
+The following [rule options](/writing-rules/rule-syntax/#options) can be used to minimize false positives:
 
-If you specify a sink such as `sink(...)` then any tainted data passed to sink, through any of its arguments, results in a match. You can narrow it down to a specific parameter this way: 
+| Rule option                   | Default | Description                                                            |
+| :-----------------------------| :------ | :--------------------------------------------------------------------- |
+| `taint_assume_safe_booleans`  | `false` | Boolean data is never considered tainted (works better with type annotations). |
+| `taint_assume_safe_numbers`   | `false` | Numbers (integers, floats) are never considered tainted (works better with type annotations). |
+| `taint_assume_safe_indexes`   | `false` | An index expression `I` tainted does not make an access expression `E[I]` tainted (it is only tainted if `E` is tainted). |
+| `taint_assume_safe_functions` | `false` | A function call like `F(E)` is not considered tainted even if `E` is tainted. (When using Pro's [inter-procedural taint analysis](#inter-procedural-analysis-pro), this only applies to functions for which Semgrep cannot find a definition.)  |
+| `taint_only_propagate_through_assignments` 🧪 | `false` | Disables all implicit taint propagation except for assignments. |
 
-```javascript
-pattern-sinks:
-  - patterns:
-    - pattern-inside: $S = new Sandbox(); ...
-    - pattern: $S.run($SINK, ...)
-    - focus-metavariable: $SINK
-```
+### Restrict taint by type (Pro)
 
-This rule causes Semgrep to only annotate the first parameter passed to `$S.run` as the sink, rather than the method `$S.run` itself. If taint goes into any other parameter of `$S.run`, then that is not considered a problem.
+By enabling `taint_assume_safe_booleans` Semgrep will automatically sanitize Boolean expressions (when it can infer that).
 
-Remember, anything can be a sink, even the index of an array access:
+For example, comparing a tainted string against a constant string will not be considered a tainted expression:
 
-```javascript
-pattern-sinks:
-  - patterns:
-    - pattern-inside: $ARRAY[$SINK]
-    - focus-metavariable: $SINK
-```
+<iframe src="https://semgrep.dev/embed/editor?snippet=6JvzK" border="0" frameBorder="0" width="100%" height="432"></iframe>
 
-This way we tell Semgrep that we do not want arrays to be accessed with tainted indexes.
+Similarly, enabling `taint_assume_safe_numbers` Semgrep will automatically sanitize numeric expressions (when it can infer that).
```

Comment:
```suggestion
Similarly, enabling `taint_assume_safe_numbers` Semgrep will automatically sanitize numeric expressions when it can infer that the expression is numeric.
```